core/src/main/java/org/elasticsearch/action/support/AutoCreateIndex.java

```diff
@@ -67,11 +69,13 @@ public boolean needToCheck() {
     /**
      * Should the index be auto created?
      */
-    public boolean shouldAutoCreate(String index, ClusterState state) {
+    public boolean shouldAutoCreate(String index, ClusterState state, IndexNameExpressionResolver resolver) {
         if (!needToCheck) {
             return false;
         }
-        if (state.metaData().hasConcreteIndex(index)) {
+        IndicesOptions options = IndicesOptions.fromOptions(true, true, true, true);
+        String[] concreteIndices = resolver.concreteIndices(state, options, index);
+        if (concreteIndices.length > 0) {
```

Comment:
ok, the concreteIndices call here scares me a bit... maybe we could have a more lightweight hasConcreteIndex without really expanding here, although index name expressions do need to be expanded? I think we should probably try and do only what we need to do , without indices options if possible... do you think it is possible?